<!--
 * @Descripttion: 
 * @version: 
 * @Author: by_mori
 * @Date: 2021-09-25 20:38:09
 * @LastEditors: by_mori
 * @LastEditTime: 2021-09-25 20:44:27
-->
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>await</title>
  </head>
  <body>
    <script>
      async function main() {
        let p = new Promise((resolve, reject) => {
          //   resolve('ok');
          reject('errorrrrr');
        });
        //1. 右侧为promise的情况
        // let res = await p;

        //2. 右侧为其他类型的数据

        // let res2 = await 20;

        //3. 如果promise是失败的状态
        try {
          let res3 = await p;
        } catch (error) {
          console.log(error);
        }

        // console.log(res);
        // console.log(res2);
        // console.log(res3);
      }
      main();
    </script>
  </body>
</html>
